### PR TITLE
Move outbound message handling to OutboundHandler

### PR DESCRIPTION
**Pull request by Tim-Brooks**
Currently there are some components of message serializer and sending
that still occur in `TcpTransport`. This commit makes it possible to
send a message without the `TcpTransport` by moving all of the remaining
application logic to the `OutboundHandler`. Additionally, it adds unit
tests to ensure that this logic works as expected.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/packaging-sample

File: server/src/main/java/org/elasticsearch/transport/DelegatingTransportMessageListener.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+package org.elasticsearch.transport;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+final class DelegatingTransportMessageListener implements TransportMessageListener {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/transport/DelegatingTransportMessageListener.java]: This is just `org.elasticsearch.transport.TransportService.DelegatingTransportMessageListener` an we should delete it from `TransportService` then?

Comment (tbrooks8) [reply server/src/main/java/org/elasticsearch/transport/DelegatingTransportMessageListener.java]: I deleted the stand alone class. It was from an unrelated change.

**Review by original-brownbear — APPROVED** [review]
thanks @tbrooks8 LGTM

**Pull request merged**